modules/structs/hook.go

```diff
@@ -91,10 +91,11 @@ type PayloadCommit struct {
 
 // PayloadCommitVerification represents the GPG verification of a commit
 type PayloadCommitVerification struct {
-	Verified  bool   `json:"verified"`
-	Reason    string `json:"reason"`
-	Signature string `json:"signature"`
-	Payload   string `json:"payload"`
+	Verified  bool         `json:"verified"`
+	Reason    string       `json:"reason"`
+	Signature string       `json:"signature"`
+	Signer    *PayloadUser `json:"signer"`
```

Comment:
Why is `Signer` needed?

Reply:
The user signing a commit is not necessarily the author or committer.

Reply:
Yes but does it really matter? `Signature` is pretty much what we need for verification

Reply:
It will give a reference the signers name/email address (if we can match it) - so you can look them up in a keyserver or on Gitea. 

A GPG signature only contains the ID of the key that signed the signature - if you don't have a public key in your local keyring that matches that ID you can't verify the signature. 

Now if Gitea has verified the signature you can figure out how it has verified the signature.

Reply:
But there is only 3 keys that it can be - Gitea server signing key, repo singing key or commiter signing key

Reply:
Could be anyone. You can sign anyone's commit.

Reply:
But it won't be verified then imho

Reply:
There is no parsing of the git commit when verifying a commit - it is simply does a key with ID that matches the ID of the signature decrypt the signature to the commit. 

The key that matches the signature does not need to match anyone on the commit. It simply has to be a key on the keyring.

An example:

* Take a copy of the git repository
* `git checkout 08da6496b61341ec45eac36afcc8f94242763468 -b re-sign`
* `git cat-file commit re-sign` - Pay attention to the Committer
```
tree 27fee9c50528ef1b0960c2c342ff139e36ce2076
parent 07f25ad8b235c2682a36ee76d59734ec3f08c413
author Junio C Hamano <gitster@pobox.com> 1570770961 +0900
committer Junio C Hamano <gitster@pobox.com> 1570771489 +0900

Eighth batch

Signed-off-by: Junio C Hamano <gitster@pobox.com>
```
* `GIT_COMMITTER_NAME="Junio C Hamano" GIT_COMMITTER_EMAIL="gitster@pobox.com" git commit --amend -S --no-edit`
* `git cat-file commit re-sign`
```
tree 27fee9c50528ef1b0960c2c342ff139e36ce2076
parent 07f25ad8b235c2682a36ee76d59734ec3f08c413
author Junio C Hamano <gitster@pobox.com> 1570770961 +0900
committer Junio C Hamano <gitster@pobox.com> 1571173132 +0100
gpgsig -----BEGIN PGP SIGNATURE-----
 
 iQIzBAABCgAdFiEE0s922pXyAemQFTKrPN50Yx8Tp0gFAl2mMw4ACgkQPN50Yx8T
 p0gnjA/+KVFqjIeemZmcmDeNlAbyBaytOV3GEEUpYDWhqGah7gQpizcReE7MjbY6
 x0MHhYygzgCa7JnCAZdAVTWqRiUwRceOLiA2X+oL9dJMCdBZ5H0SW9B1BAduPk6S
 AutRbXjFnPggTS3pKdttqhIIws/FPaGxVAwBxTP0tPICCEARF+Q070Huzz7z4bMU
 rUiLQvEs7aQ4X6PRBP+/84911XhXMOJSDAMJQBQHTA/avLMXY+QT7zsam+nDH1ah
 7ZTi83AIMTUiBhIafJtEPKF7w8a5hP68x75Pcdh9zCamVOKNrcVZA3/UjcBr4jNO
 9+ho0CFA/RguRUPUJg8/XsIBcqtKBE+sPL3GwhD5w6cSX6xMj59ToCqxjmLMtR/4
 fQ6vQTh2ib8lJktb78n/VN7yOYYXMKyYocSgFzjtzT62+TAXP5uqQtCz6QSiX9l9
 mZIvLlYQyX/T7/HkV7PLCLZo0fgMHin/1VRVXeI0yA3R9aoZ+MN2PWXEUbfSzkjF
 bc1XxLr7ihm3CLEpSecVmXZNXwnkVj355/tPNf+AMIAjJFls2lZSW3p56xDvWBfa
 Z6XTD4V2ivWHbWRftA/8jknxEkIS1HPIrzXZVYujuAlH8NWRgqpPXtdVfNpf2ZZE
 JXkuSb059hF+brZVryfOGiZuIAac0kmRIjcx+kQryN+mKIgl0Cg=
 =tTjx
 -----END PGP SIGNATURE-----

Eighth batch

Signed-off-by: Junio C Hamano <gitster@pobox.com>
```
* `git verify-commit re-sign` 
```
gpg: Signature made Tue 15 Oct 2019 21:58:54 BST
gpg:                using RSA key D2CF76DA95F201E9901532AB3CDE74631F13A748
gpg: Good signature from "Andrew Thornton <art27@cantab.net>" [ultimate]
gpg: Signature made Tue 15 Oct 2019 21:58:54 BST
gpg:                using RSA key D2CF76DA95F201E9901532AB3CDE74631F13A748
gpg: Good signature from "Andrew Thornton <art27@cantab.net>" [ultimate]
```

Reply:
@lafriks The signer can change, update, etc. their gpg key after signing. I think it's valuable to keep the actual key instead of the signer identity.

Reply:
(And of course, @zeripath 's explanation)

Reply:
So, yeah, when I discovered this I was quite surprised, but it allows you to re-sign someone else's trees affirming your continued trust in those commits even if you no longer trust their signature. 

Unfortunately this changes the sha of the commit so you have to effectively rebase by doing a git filter-branch for each commit and if you do do it because a key was compromised and is no longer trustworthy you have you change your heads to use the new commit heads. 

(However the tree SHA should be the same so there should be a way to get git to recognise the commits would apply.)